### PR TITLE
Avoid local copy of decomp_info objects

### PR DESCRIPTION
**Pull request by mathrack**

### CODE REVIEW

**Review by pbartholomew08 — CHANGES_REQUESTED** [review]
See above discussions

File: src/fft_common.f90 [hunk 1]
```diff
@@ -27,10 +27,11 @@
 integer, save :: nx_fft, ny_fft, nz_fft
 
 ! 2D processor grid
+! FIXME this is already available in the module decomp_2d
 integer, save, dimension(2) :: dims
 
 ! Decomposition objects
-TYPE(DECOMP_INFO), save :: ph  ! physical space
+TYPE(DECOMP_INFO), pointer, save :: ph=>null()  ! physical space
```
[the remark this answers is not in the document]
Comment (mathrack) [reply src/fft_common.f90:34]: This modification is limited to the FFT engine. User code do not have access to this decomp_info object.

File: src/fft_common.f90 [hunk 2]
```diff
@@ -112,6 +113,7 @@ subroutine fft_init_general(pencil, nx, ny, nz)
 nz_fft = nz
 
 ! determine the processor grid in use
+! FIXME this is already defined in the module decomp_2d
```
Comment (mathrack) [inline src/fft_common.f90:116]: This FIXME should be in another PR, no fix is proposed here.

File: src/fft_common.f90 [hunk 3]
```diff
@@ -27,10 +27,11 @@
 integer, save :: nx_fft, ny_fft, nz_fft
 
 ! 2D processor grid
+! FIXME this is already available in the module decomp_2d
```
Comment (mathrack) [inline src/fft_common.f90:30]: This FIXME should be in another PR, no fix is proposed here.

File: src/fft_fftw3_f03.f90 [hunk 4]
```diff
@@ -52,10 +52,11 @@ module decomp_2d_fft
   integer, save :: nx_fft, ny_fft, nz_fft
 
   ! 2D processor grid
+  ! FIXME this is already available in the module decomp_2d
```
Comment (mathrack) [inline src/fft_fftw3_f03.f90:55]: This FIXME should be in another PR, no fix is proposed here.

File: src/decomp_2d.f90 [hunk 5]
```diff
@@ -654,12 +672,17 @@ end subroutine decomp_mpi_comm_free
 !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
   ! Return the default decomposition object
 !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+  ! FIXME avoid a copy and return a pointer to decomp_main
```
[the remark this answers is not in the document]
Comment (pbartholomew08) [reply src/decomp_2d.f90:676]: ok this should be an obvious change to any code users

[the remark this answers is not in the document]
Comment (mathrack) [reply src/decomp_2d.f90:676]: If they have a pointer to the object decom_main, they have the power to finalize it. With great power comes great responsibility.

**Comment by mathrack** [comment]
Any update needed in this PR ?

**Review by pbartholomew08 — APPROVED** [review]

**Pull request merged**